source/config/src/lib.rs

```diff
@@ -0,0 +1,15 @@
+#![cfg_attr(not(any(feature = "use-std", test)), no_std)]
+
+use core::marker::PhantomData;
+
+use serde::{Deserialize, Serialize};
+
+pub struct MnemosConfig<'a, K, P>
+where
+    K: Serialize + Deserialize<'a>,
+    P: Serialize + Deserialize<'a>,
```

Comment:
hmm, it seems like, if we were to get rid of the bounds on the struct itself, we wouldn't need the `PhantomData` for the deserializer lifetime? Instead, we could just have the bounds on `impl` blocks for this struct, and then we wouldn't need a `PhantomData`.